### PR TITLE
feat(assets): Bulk mileage registration

### PR DESCRIPTION
**Pull request by thorkellmani**
## What

* Add a bulk mileage registration service
* new pages for bulk mileage registrations

## Why

*Owners of multiple vehicles should be able to register mileages in bulk

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
	- Introduced new input types for bulk vehicle mileage requests and status.
	- Added new constants for origin code and logging category.
	- Implemented new interfaces for mileage readings and vehicle data structures.
	- Added support for bulk mileage registration jobs with detailed response handling.
	- New React components for managing bulk mileage input, job overview, and file downloads.
	- Enhanced localization with Icelandic translations for vehicle-related messages.
	- Introduced new SVG icons for upload actions.

- **Bug Fixes**
	- Removed outdated vehicle and user vehicle data structures.

- **Documentation**
	- Enhanced GraphQL schemas for better data handling and clarity.

### CODE REVIEW

**Review by baering — APPROVED** [review]
Core file LGTM

File: libs/api/domains/vehicles/src/lib/resolvers/vehicleV3.resolver.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  })
+  @Audit()
+  async vehicleMileageRegistrations(
+    @CurrentUser() user: User,
+    @Args('input', { nullable: true }) input: GetVehicleMileageInput,
+  ) {
+    return this.vehiclesService.getVehicleMileageHistory(user, input)
+  }
+
+  //field resolve mileage registrations?
```
Comment (thordurhhh) [inline libs/api/domains/vehicles/src/lib/resolvers/vehicleV3.resolver.ts:49]: Remove?

File: libs/api/domains/vehicles/src/lib/resolvers/mileage.resolver.ts [hunk 2]
```diff
@@ -90,7 +98,7 @@ export class VehiclesMileageResolver {
 
     if (!res) return undefined
 
-    return mileageDetailConstructor(res)
+    return mileageDetailConstructor(res[0])
```
[the remark this answers is not in the document]
Comment (thordurhhh) [reply libs/api/domains/vehicles/src/lib/resolvers/mileage.resolver.ts:93]: @thorkellmani - hefur 🐰 ekki eitthvað til síns máls hérna?

File: libs/service-portal/assets/src/screens/VehicleBulkMileageUpload/VehicleBulkMileageUpload.tsx [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    if (id && id !== requestGuid) {
+      setRequestGuid(id)
+    }
+  }, [data?.vehicleBulkMileagePost?.requestId])
+
+  const postMileage = async (file: File) => {
+    try {
+      const records = await parseCsvToMileageRecord(file)
+      if (!records.length) {
+        setUploadErrorMessage('Engin gild kílómetrastaða fannst í skjali')
```
Comment (thordurhhh) [inline libs/service-portal/assets/src/screens/VehicleBulkMileageUpload/VehicleBulkMileageUpload.tsx:54]: Translation string needed?

File: libs/service-portal/assets/src/screens/VehicleBulkMileageUpload/VehicleBulkMileageUpload.tsx [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+              mileageNumber: r.mileage,
+              vehicleId: r.vehicleId,
+            })),
+            originCode: 'ISLAND.IS',
+          },
+        },
+      })
+    } catch (error) {
+      setUploadErrorMessage(
+        'Error occurred while processing the file: ' + error.message,
```
Comment (thordurhhh) [inline libs/service-portal/assets/src/screens/VehicleBulkMileageUpload/VehicleBulkMileageUpload.tsx:70]: Translation needed?

**Review by thordurhhh — APPROVED** [review]
Nice 🙌

**Comment by thorkellmani** [comment]
@coderabbitai review pls

File: libs/island-ui/core/src/lib/IconRC/icons/UploadOutline.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+import type { SvgProps as SVGRProps } from '../types'
+
+const SvgUploadOutline = ({
+  title,
+  titleId,
+  ...props
+}: React.SVGProps<SVGSVGElement> & SVGRProps) => {
+  return (
+    <svg
+      className="download-outline_svg__ionicon"
```
Comment (disaerna) [inline libs/island-ui/core/src/lib/IconRC/icons/UploadOutline.tsx:11]: className is download, this should prob be upload right?

File: libs/island-ui/core/src/lib/IconRC/icons/UploadOutline.tsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+      viewBox="0 0 512 512"
+      aria-labelledby={titleId}
+      transform="matrix(1,0,0,-1,0,0)"
```
Comment (disaerna) [inline libs/island-ui/core/src/lib/IconRC/icons/UploadOutline.tsx:14]: this has transform matrix and the other is rotate 180, why is that 💭 ?

**Review by disaerna — APPROVED** [review]
UI Core files reviewed - only minor questions

File: libs/island-ui/core/src/lib/IconRC/icons/UploadOutline.tsx (hunk 5 above)
Comment (thorkellmani) [reply libs/island-ui/core/src/lib/IconRC/icons/UploadOutline.tsx:11]: omg yeah ofc

File: libs/island-ui/core/src/lib/IconRC/icons/UploadOutline.tsx (hunk 6 above)
Comment (thorkellmani) [reply libs/island-ui/core/src/lib/IconRC/icons/UploadOutline.tsx:14]: coderabbit review fix mixup 🙈

**Comment by saevarma** [comment]
Removed `automerge` to unblock Kodiak

**Pull request merged**